test/bioprinttest.c

```diff
@@ -268,16 +262,21 @@ void test_close_streams(void)
 
 int test_puts_stdout(const char *str)
 {
-    return write(1, str, strlen(str));
+    return fputs(str, stdout);
 }
 
 int test_puts_stderr(const char *str)
 {
-    return write(2, str, strlen(str));
+    return fputs(str, stderr);
 }
 
 static char vprint_buf[10240];
 
+/*
+ * This works out as long as caller doesn't use any "fancy" formats.
+ * But we are caller's caller, and test_str_eq is the only one called,
+ * and it uses only "%s", which is not "fancy"...
+ */
```

Comment:
The only "fancy" ones to date are `TEST_size_t_eq` / `TEST_size_t_ne`.  Do note the definition of `TESTUTIL_NO_size_t_COMPARISON` at the beginning of this file and what that does in `testutil.h`